wsmaster/che-core-sql-schema/src/main/resources/che-schema/6.16.0/2__create_workspace_activity_table.sql

```diff
@@ -0,0 +1,31 @@
+--
+-- Copyright (c) 2012-2018 Red Hat, Inc.
+-- This program and the accompanying materials are made
+-- available under the terms of the Eclipse Public License 2.0
+-- which is available at https://www.eclipse.org/legal/epl-2.0/
+--
+-- SPDX-License-Identifier: EPL-2.0
+--
+-- Contributors:
+--   Red Hat, Inc. - initial API and implementation
+--
+
+CREATE TABLE che_workspace_activity (
+    workspace_id               VARCHAR(255) NOT NULL,
+    status                     VARCHAR(255),
+    created                    BIGINT,
+    last_starting              BIGINT,
+    last_running               BIGINT,
+    last_stopping              BIGINT,
+    last_stopped               BIGINT,
+    expiration                 BIGINT,
+
+    PRIMARY KEY (workspace_id)
+);
+ALTER TABLE che_workspace_activity ADD CONSTRAINT ws_activity_workspace_id FOREIGN KEY (workspace_id) REFERENCES workspace (id);
+CREATE INDEX che_index_ws_activity_status ON che_workspace_activity (status);
+CREATE INDEX che_index_ws_activity_last_starting ON che_workspace_activity (last_starting);
```

Comment:
You don't have queries that search activities by `last_starting` only, but you have a query that search by `status + last_starting`. Are you sure that such index is enough or maybe `che_workspace_activity (status, last_starting);` should we have instead. Feel free to ask if you need instruction how to check it.

Reply:
Agreed, I didn't touch these as I evolved the queries.